api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -908,12 +911,9 @@ private static String generateNotebookUserCode(
     String namespace = "dataset_" + qualifier + "_" + domainAsString + "_";
     // Comments in R and Python have the same syntax
     String descriptiveComment =
-        new StringBuilder("# This query represents dataset \"")
-            .append(dataSetName)
-            .append("\" for domain \"")
-            .append(domainAsString)
-            .append("\"")
-            .toString();
+        String.format(
+            "# This query represents dataset \"%s\" for domain \"%s\" and was generated for %s",
```

Comment:
Scope creep, but I would recommend just using a JSON blob in the comment, since it's easier for us to scan later if we need to, and we can .add fields more easily without ti becoming unwieldy to change. 

Or maybe we add an invisible `.aou/queryManifest.json` to the bucket and do this there.

I'm just really uncomfortable embedding such time-sensitive information in a user-writable place.

Reply:
Discussed offline, and I agree this is one starting point, but I'm still wary of embedding metadata casually like this in a user-editable and non-scannable format. We can use a regex to see things like this if we're scanning a notebook, but that means the pattern is a contract.